scripts/node4.sh

```diff
@@ -0,0 +1,213 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+echo "Running node4.sh"
+
+adminUsername=$1
+adminPassword=$2
+uniqueString=$3
+location=$4
+graphDatabaseVersion=$5
+installGraphDataScience=$6
+graphDataScienceLicenseKey=$7
+installBloom=$8
+bloomLicenseKey=$9
+nodeCount=${10}
+loadBalancerDNSName=${11}
+azLoginIdentity=${12}
+resourceGroup=${13}
+vmScaleSetsName=${14}
+
+echo "Turning off firewalld"
+systemctl stop firewalld
+systemctl disable firewalld
+
+#Format and mount the data disk to /var/lib/neo4j
+MOUNT_POINT="/var/lib/neo4j"
+
+DATA_DISK_DEVICE=$(parted -l 2>&1 | grep Error | awk {'print $2'} | sed 's/\://')
+
+sudo parted $DATA_DISK_DEVICE --script mklabel gpt mkpart xfspart xfs 0% 100%
+sudo mkfs.xfs $DATA_DISK_DEVICE\1
+sudo partprobe $DATA_DISK_DEVICE\1
+mkdir $MOUNT_POINT
+
+DATA_DISK_UUID=$(blkid | grep $DATA_DISK_DEVICE\1 | awk {'print $2'} | sed s/\"//g)
+
+echo "$DATA_DISK_UUID $MOUNT_POINT xfs defaults 0 0" >> /etc/fstab
+
+systemctl daemon-reload
+mount -a
+
+
+install_azure_from_dnf() {
+
+  sudo rpm --import https://packages.microsoft.com/keys/microsoft.asc
+  sudo dnf install -y https://packages.microsoft.com/config/rhel/8/packages-microsoft-prod.rpm
+echo -e "[azure-cli]
+name=Azure CLI
+baseurl=https://packages.microsoft.com/yumrepos/azure-cli
+enabled=1
+gpgcheck=1
+gpgkey=https://packages.microsoft.com/keys/microsoft.asc" | sudo tee /etc/yum.repos.d/azure-cli.repo
+sudo dnf install -y azure-cli
+}
+
+perform_az_login() {
+  echo "Performing az login"
+  az login --identity -u "${azLoginIdentity}"
+}
+
+install_neo4j_from_yum() {
+
+  echo "Adding neo4j yum repo..."
+  rpm --import https://debian.neo4j.com/neotechnology.gpg.key
+  echo "
+[neo4j]
+name=Neo4j Yum Repo
+baseurl=http://yum.neo4j.com/stable/4.4
+enabled=1
+gpgcheck=1" > /etc/yum.repos.d/neo4j.repo
+
+  echo "Installing Graph Database..."
+  export NEO4J_ACCEPT_LICENSE_AGREEMENT=yes
+  set_yum_pkg
+  yum -y install "${yumPkg}"
+  systemctl enable neo4j
+}
+
+install_apoc_plugin() {
+  echo "Installing APOC..."
+  mv /var/lib/neo4j/labs/apoc-*-core.jar /var/lib/neo4j/plugins
+}
+
+get_latest_neo4j_version() {
+  echo "Getting latest neo4j version"
+  latest_neo4j_version=$(curl -s --fail http://versions.neo4j-templates.com/target.json | jq -r '.azure."4.4"' || echo "")
+  echo "Latest Neo4j Version is ${latest_neo4j_version}"
+}
+
+get_vmss_tags() {
+  taggedNeo4jVersion=$(az vmss list --resource-group ${resourceGroup} | jq --arg vmssName "${vmScaleSetsName}" '.[] | select(.name==$vmssName).tags.Neo4jVersion')
+  echo "Tagged Neo4j Version ${taggedNeo4jVersion}"
+}
+
+set_vmss_tags() {
+  installed_neo4j_version=$(/usr/bin/neo4j --version)
```

Comment:
consider `local -r`

Reply:
this is required outside the func hence cannot be local